sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -121,30 +121,47 @@ Use ``node make help`` to list all available commands.
 You can use ``tsc`` directly for error checking (e.g. in an IDE). However, don't use
 it for code emit, because we rely on AST transforms to produce viable library code.
 
+.. _devguide_bokehjs_development_testing:
+
 Testing
 ~~~~~~~
 
-BokehJS testing is performed with the ``node make test`` command. You can run individual
-test suites with ``node make test:suite_name``. Known tests suites are:
+BokehJS comes with its own suites of tests and testing framework. All tests for BokehJS
+use ``describe()`` and ``it()`` functions. Unit tests are written using Chai "expect"
+style. [TBD: How are style/setup different for visual integration tests?]
+
+To launch BokehJS tests, run ``node make test`` from within the
+:bokeh-tree:`bokehjs/test` directory.
+
+Instead of running all available BokehJS tests, you can also run individual test
+suites with ``node make test:suite_name``. Available tests suites are:
+
+* ``node make test:codebase``: Codebase tests checking file size limits
+* ``node make test:defaults``: Tests checking whether the defaults in Bokeh's
+  Python models match those of Bokeh's TypeScript models
+* ``node make test:unit``: Unit tests for BokehJS
+* ``node make test:integration``:
```

Comment:
@mattpap Are there any BokehJS integration tests that are not visual tests? I.e. are the visual tests a subset of the integration tests, or would it be fair to say all BokehJS integration tests are visual tests?

Reply:
I think it is safe to say that all BokehJS integration tests are visual tests.

Reply:
Yes, I concur.  If I run `node make test:integration` then I see
```
████████████████████████████████████████ 100% | 584 of 584 | 88 failures | 1 skipped
```
so with one skipped test there are 583 tests run. Total number of linux baseline images is
```
> ls test/baselines/linux/*png  | wc -l
583
```

Reply:
Thanks!